src/meshlink.c

```diff
@@ -1807,6 +1851,25 @@ int meshlink_get_node_dev_class(meshlink_handle_t *mesh, meshlink_node_t *node)
 	return devclass;
 }
 
+meshlink_submesh_t *meshlink_get_node_submesh(meshlink_handle_t *mesh, meshlink_node_t *node) {
+	if(!mesh || !node) {
+		meshlink_errno = MESHLINK_EINVAL;
+		return NULL;
+	}
+
+	node_t *n = (node_t *)node;
+
+	meshlink_submesh_t *s;
+
+	pthread_mutex_lock(&(mesh->mesh_mutex));
```

Comment:
Since a node is never allowed to change to a different sub-mesh, you don't need to take a lock here.

Reply:
Lock has been removed in the latest changes